test/test_fmcomms2-3_p.py

```diff
@@ -0,0 +1,216 @@
+            150,
+            dict(
+                sample_rate=30720000,
+                tx_lo=2300000000,
+                rx_lo=2400000000,
+                gain_control_mode_chan0="slow_attack",
+                gain_control_mode_chan1="slow_attack",
+                rx_rf_bandwidth=18000000,
+                tx_rf_bandwidth=18000000,
+            ),
+        ),
+        (
+            0.4,
+            10,
+            50,
+            dict(
+                gain_control_mode_chan0="slow_attack",
+                gain_control_mode_chan1="slow_attack",
+                rx_lo=2400000000,
+                tx_lo=2400000000,
+                tx_hardwaregain_chan0=-10,
+                tx_hardwaregain_chan1=-10,
+                sample_rate=30720000,
+                rx_rf_bandwidth=18000000,
+                tx_rf_bandwidth=18000000,
+            ),
+        ),
+    ],
+)
+def test_rssi(
+    test_gain_check,
+    iio_uri,
+    classname,
+    channel,
+    param_set,
+    dds_scale,
+    min_rssi,
+    max_rssi,
+):
+    test_gain_check(
+        iio_uri, classname, channel, param_set, dds_scale, min_rssi, max_rssi
+    )
+
+
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize("channel", [0, 1])
+@pytest.mark.parametrize(
+    "dds_scale, frequency, hardwaregain_low, hardwaregain_high",
+    [(0.0, 999859, 50, 80), (0.4, 999859, 0.0, 28)],
+)
+def test_hardware_gain(
+    test_hardwaregain,
+    iio_uri,
+    classname,
+    channel,
+    dds_scale,
+    frequency,
+    hardwaregain_low,
+    hardwaregain_high,
+):
+    test_hardwaregain(
+        iio_uri,
+        classname,
+        channel,
+        dds_scale,
+        frequency,
+        hardwaregain_low,
+        hardwaregain_high,
+    )
+
+
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize("channel", [0, 1])
+def test_ad9361_loopback(test_dma_loopback, iio_uri, classname, channel):
+    test_dma_loopback(iio_uri, classname, channel)
+
+
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize("channel", [0, 1])
+@pytest.mark.parametrize(
+    "param_set",
+    [
+        dict(
+            sample_rate=30720000,
+            tx_lo=2400000000,
+            rx_lo=2400000000,
+            gain_control_mode_chan0="slow_attack",
+            gain_control_mode_chan1="slow_attack",
+            tx_hardwaregain_chan0=-10,
+            tx_hardwaregain_chan1=-10,
+            rx_rf_bandwidth=18000000,
+            tx_rf_bandwidth=18000000,
+        ),
+    ],
+)
+def test_ad9361_iq_loopback(test_iq_loopback, iio_uri, classname, channel, param_set):
+    test_iq_loopback(iio_uri, classname, channel, param_set)
+
+
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+def test_dcxo(test_dcxo_calibration, classname, iio_uri):
+    test_dcxo_calibration(classname, iio_uri)
```

Comment:
Can we implement a pytest.skip somewhere so if the instrument is missing users get notified? pytest will hide prints if not using the -s flag.

Reply:
Thank you for the observation, I replaced the print with pytest.skip in dcxo_calibration and it works just fine when i turn off the device